File: pat(advanced)C++/1023(advanced).cpp
#include<iostream>
using namespace std;
#include<unordered_map>
unordered_map<int,int>digit[3];



int main(){
	string number;
	string number1;
	int j;
	int jinwei = 0;
	cin>>number;
	for(int i = 0;i<10;i++){
		digit[1][i] = 0;
		digit[2][i] = 0;
	}
	for(int i = 0;i<(int)number.size();i++){
		digit[1][number[i]-'0']++;
		digit[2][number[i]-'0']++;
	}
	for(int i = (int)number.size()-1;i>=0;i--){
		digit[2][number[i]-'0']--;
		j = (number[i]-'0')*2;
		j+=jinwei;
		if(j>=10){
			digit[2][j-10]++;
			number1 += '0'+j-10;
			jinwei = 1;
		}
		else{
			digit[2][j]++;
			number1 += '0'+j;
			jinwei = 0;
		}
	}
	int k;
	for(k = 0;k<10;k++){
		if(digit[1][k]!=digit[2][k]){
			break;
		}
	}
	if(k==10){
		cout<<"Yes"<<endl;
	}
	else{
		cout<<"No"<<endl;
		if(number[0]>='5'){
			cout<<'1';
		}
	}
	for(int i = (int)number.size()-1;i>=0;i--){
			cout<<number1[i];
		}
	return 0;
}

File: pat(advanced)python/1023(advanced).py
if __name__ == '__main__':
    number = input()
    number1 = str(2*int(number))
    outputstring = 'No'
    if number.count('1')==number1.count('1'):
        if number.count('2')==number1.count('2'):
            if number.count('3') == number1.count('3'):
                if number.count('4') == number1.count('4'):
                    if number.count('5') == number1.count('5'):
                        if number.count('6') == number1.count('6'):
                            if number.count('7') == number1.count('7'):
                                if number.count('8') == number1.count('8'):
                                    if number.count('9') == number1.count('9'):
                                        if number.count('0') == number1.count('0'):
                                            outputstring = 'Yes'
    print(outputstring)
    print(number1)